servicetest/agent/test_agent.py

```diff
@@ -93,23 +94,41 @@ def test_old_containers_cleanup(self):
             The second phase of the test is to run SC again and make sure
             it does not fail because of any files left from the previous run.
         """
+        # Create a container and kill SC-agent
         sc = Container()
         container_id = sc.start(DATA)
+        container_name = "SC-" + str(container_id)
         os.system("pkill --signal SIGKILL -f softwarecontainer-agent")
         time.sleep(0.5)
 
-        os.system("lxc-ls >> output")
-        with open("output", "r") as fh:
-            file_content = fh.readline()
-        file_content = file_content.strip()
-        assert file_content == ("SC-" + str(container_id))
-        os.remove("output")
+        skipped_container_name = "agent-servicetest"
+        # Create one that SC shouldn't touch
+        os.system("lxc-create -t busybox -n " + skipped_container_name)
```

Comment:
Working on it. The LXC bindings for Python are Python3 only, which means some refactoring and updating of the service test code.